ballerina/soap_utils.bal

```diff
@@ -1,106 +1,186 @@
+            securedEnvelope = check wssec:applySymmetricBinding(envelope, securityPolicy);
+        } else if securityPolicy is wssec:AsymmetricBindingConfig {
+            securedEnvelope = check wssec:applyAsymmetricBinding(envelope, securityPolicy);
+        } else {
+            securedEnvelope = envelope;
+        }
+    } else {
+        foreach wssec:InboundSecurityConfig policy in securityPolicy {
+            securedEnvelope = check applySecurityPolicies(policy, envelope);
+        }
+    }
+    return securedEnvelope;
+}
+
+public isolated function applyOutboundConfig(wssec:OutboundSecurityConfig outboundSecurity, xml envelope)
+    returns xml|Error {
+    xmlns "http://schemas.xmlsoap.org/soap/envelope/" as soap;
+    xml soapEnvelope = envelope;
+    do {
+        wssec:EncryptionAlgorithm? encryptionAlgorithm = outboundSecurity.decryptionAlgorithm;
+        if encryptionAlgorithm is wssec:EncryptionAlgorithm {
+            crypto:PrivateKey|crypto:PublicKey? clientPrivateKey = outboundSecurity.decryptionKey;
+            if clientPrivateKey is crypto:PrivateKey|crypto:PublicKey {
+                byte[] encData = check wssec:getEncryptedData(soapEnvelope);
+                byte[] decryptDataResult = check wssec:decryptData(encData, encryptionAlgorithm, clientPrivateKey);
+                string decryptedBody = "<soap:Body >" + check string:fromBytes(decryptDataResult) + "</soap:Body>";
+                string decryptedEnv = regexp:replace(re `<soap:Body .*>.*</soap:Body>`, soapEnvelope.toString(),
+                                                     decryptedBody);
+                soapEnvelope = check xml:fromString(decryptedEnv);
+            }
+        }
+        wssec:SignatureAlgorithm? signatureAlgorithm = outboundSecurity.signatureAlgorithm;
+        if signatureAlgorithm is wssec:SignatureAlgorithm {
+            crypto:PublicKey? serverPublicKey = outboundSecurity.verificationKey;
+            if serverPublicKey is crypto:PublicKey {
+                byte[] signatureData = check wssec:getSignatureData(soapEnvelope);
+                check wssec:verifyData((soapEnvelope/<soap:Body>/*).toString().toBytes(),
+                                       signatureData, serverPublicKey, signatureAlgorithm);
+            }
+        }
+        return soapEnvelope;
+    } on fail var e {
+        return error Error("Outbound security configurations do not match with the SOAP response. ", e.cause());
+    }
+}
+
+string path = "";
+
+public isolated function sendReceive(xml|mime:Entity[] body, http:Client httpClient, string? soapAction = (),
+                                     map<string|string[]> headers = {}, string path = "", boolean soap12 = true) returns xml|Error {
+    http:Request req;
+    if soap12 {
+        req = createSoap12HttpRequest(body, soapAction, headers);
+    } else {
+        req = createSoap11HttpRequest(body, <string>soapAction, headers);
+    }

```

Comment:
we could use a conditional statement here

Reply:
Fixed in [e309edb](https://github.com/ballerina-platform/module-ballerina-soap/pull/103/commits/e309edbbc4d7673f4c2ece8673433cf028487275)